src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,295 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager::AnalyzerManager(UserSettingsPointer pConfig) :
+    m_pConfig(pConfig),
+    m_nextWorkerId(0),
+    m_batchTrackQueue(),
+    m_prioTrackQueue(),
+    m_backgroundWorkers(),
+    m_foregroundWorkers(),
+    m_pausedWorkers() {
+
+    int maxThreads = m_pConfig->getValue<int>(ConfigKey("[Library]", "MaxAnalysisThreads"));
+    int ideal = QThread::idealThreadCount();
+    if (QThread::idealThreadCount() < 1) {
```

Comment:
Do you have test data, that the QT idealThreadCount actually is ideal in this case?